backend/cellguide/pipeline/computational_marker_genes/computational_markers.py

```diff
@@ -238,11 +162,122 @@ def _prepare_cell_counts_and_gene_expression_dfs(
+        """
+        This function is used in dask scheduler to process each cell type in parallel.
+
+        Parameters
+        ----------
+        i : int
+            The index of the current cell type being processed.
+        cell_types_o : np.ndarray
+            Array of cell types.
+        e_sum_o : np.ndarray
+            Array of gene expression sums for each cell type.
+        e_sqsum_o : np.ndarray
+            Array of squared gene expression sums for each cell type.
+        n_cells_o : np.ndarray
+            Array of cell counts for each cell type.
+        e_sum_o_orig : np.ndarray
+            Array of raw gene expression sums for each cell type.
+        e_sqsum_o_orig : np.ndarray
+            Array of raw squared gene expression sums for each cell type.
+        n_cells_o_orig : np.ndarray
+            Array of cell counts for each cell type.
+        filter_genes: np.ndarray
+            A boolean array indicating which genes to filter out.
+
+        Returns
+        -------
+        tuple[np.ndarray, np.ndarray]
+            A tuple containing the filtered effect sizes and the column indices of the remaining genes.
+        """
+        e_sum_o = e_sum_o.copy()
+        e_sqsum_o = e_sqsum_o.copy()
+        n_cells_o = n_cells_o.copy()
+        cell_type_target = cell_types_o[i]
+
+        indexer = pd.Series(index=cell_types_o, data=np.arange(cell_types_o.size))
+
+        for j, cell_type in enumerate(cell_types_o):
+            if cell_type_target == cell_type or are_cell_types_colinear(cell_type, cell_type_target):
+                continue
+
+            overlapping_descendants = get_overlapping_cell_type_descendants(cell_type, cell_type_target)
+            overlapping_descendants = list(set(overlapping_descendants).intersection(cell_types_o))
+            if len(overlapping_descendants) > 0:
+                e_sum_o[j] -= e_sum_o_orig[indexer[overlapping_descendants].values].sum(0)
+                e_sqsum_o[j] -= e_sqsum_o_orig[indexer[overlapping_descendants].values].sum(0)
+                n_cells_o[j] -= n_cells_o_orig[indexer[overlapping_descendants].values].sum(0)
+
+        # get the expressions and cell counts corresponding to the cell type
+        sum1 = e_sum_o[i][None, :]
+        sumsq1 = e_sqsum_o[i][None, :]
+        n1 = n_cells_o[i][None, :]
+
+        # calculate cohens d effect size against all other rows
+        effects = calculate_cohens_d(sum1=sum1, sumsq1=sumsq1, n1=n1, sum2=e_sum_o, sumsq2=e_sqsum_o, n2=n_cells_o)
+
+        # zero out nans
+        effects[np.isnan(effects)] = 0
+
+        # get valid genes
+        unique_cols = np.where(~filter_genes)[0]
+
+        # filter out rows that are colinear with the cell type target and filter out invalid genes
+        is_colinear = np.array([are_cell_types_colinear(cell_type, cell_type_target) for cell_type in cell_types_o])
+        effects_sub = effects[~is_colinear][:, unique_cols]
```

Comment:
Get the effect sizes corresponding to non-colinear comparisons and valid genes (nnz > 25)